sqld/src/postgres/handlers.rs

```diff
@@ -98,47 +94,85 @@ impl ExtendedQueryHandler for QueryHandler {
+                    )
+                })?
+                .unwrap_or_default();
+
+            let params = parse_params(portal.statement().parameter_types(), portal.parameters());
+
+            let query = Query { stmt, params };
+            let response = self.handle_queries(vec![query]).await.map(|mut res| {
+                assert_eq!(res.len(), 1);
+                res.pop().unwrap()
+            })?;
+            match response {
+                Response::EmptyQuery => {
+                    client
+                        .feed(PgWireBackendMessage::EmptyQueryResponse(EmptyQueryResponse))
+                        .await?;
+                }
+                Response::Query(results) => {
+                    let include_col_defs = client.metadata_mut().remove(REQUEST_DESCRIBE).is_some();
+                    send_query_response(client, results, include_col_defs).await?;
+                }
+                Response::Execution(tag) => {
+                    client
+                        .send(PgWireBackendMessage::CommandComplete(tag.into()))
+                        .await?;
+                }
+                Response::Error(err) => {
+                    client
+                        .send(PgWireBackendMessage::ErrorResponse((*err).into()))
+                        .await?;
+                }
+            }
+            client
+                .send(PgWireBackendMessage::ReadyForQuery(ReadyForQuery::new(
+                    READY_STATUS_IDLE,
+                )))
+                .await?;
+
+            Ok(())
+        } else {
+            Err(PgWireError::PortalNotFound(portal_name.to_owned()))
+        }
+    }
+
     async fn do_query<'q, 'b: 'q, C>(
         &'b self,
-        client: &mut C,
-        portal: &'q Portal<String>,
+        _client: &mut C,
+        _portal: &'q Portal<String>,
         _max_rows: usize,
     ) -> PgWireResult<Response<'q>>
     where
         C: ClientInfo + Unpin + Send + Sync,
     {
-        debug_assert_eq!(
-            portal.statement().parameter_types().len(),
-            portal.parameter_len()
-        );
-
-        let stmt = Statement::parse(portal.statement().statement())
-            .next()
-            .transpose()
-            .map_err(|e| {
-                PgWireError::UserError(
-                    ErrorInfo::new("ERROR".into(), "XX000".into(), e.to_string()).into(),
-                )
-            })?
-            .unwrap_or_default();
-
-        let params = parse_params(portal.statement().parameter_types(), portal.parameters());
-
-        let query = Query { stmt, params };
-        let include_col_defs = client.metadata_mut().remove(REQUEST_DESCRIBE).is_some();
-        self.handle_queries(vec![query], include_col_defs)
-            .await
-            .map(|mut res| {
-                assert_eq!(res.len(), 1);
-                res.pop().unwrap()
-            })
+        unreachable!()
```

Comment:
I get that this is unreachable because we provide an implementation for `on_execute`?

Reply:
Yes. This is called by `on_execute`'s default implementation. However, we now use a custom implementation because of how we deal with response of `Describe` message. So `do_query` is no longer called.